control-plane/agents/src/bin/core/controller/registry.rs

```diff
@@ -234,17 +240,33 @@ impl Registry {
         }
     }
 
+    async fn op_with_threshold<F, O>(future: F) -> O
+    where
+        F: Future<Output = O>,
+    {
+        let start = std::time::Instant::now();
+        let result = future.await;
+        let warn_threshold = std::time::Duration::from_secs(1);
```

Comment:
Should the threshold be more, is 1s too fast ? How did we calculate this threshold?